lib/contracts/shared-contracts.ts

```diff
@@ -9,6 +9,12 @@ export namespace SharedContracts {
         codename?: string;
         external_id?: string;
     }
+
+    export interface IUserReferenceObjectContract {
```

Comment:
I don't think this type is enough - I would like to see one type fro get
```ts
// or something like this
export interface IGetUserRefrenceObjectContract  { 
    id: string 
}
```

![image](https://github.com/user-attachments/assets/e7850de0-6b3e-46a8-a8a6-0690af74bf33)


and one type union for put 
```ts
export type IPutUserRefrenceObjectContract = Readonly<{ 
id?: string 
email: string
}  | {
id:string,
email? :string
}>
```

![image](https://github.com/user-attachments/assets/b73b99ae-11b5-4e8b-9edc-6078e54a600a)

Reply:
I've improved the models as per recommendation :-) I've only changed the naming and used `never` instead of using just `undefined` because this way you can't use both `id` & `email` in POST / PUT requests and have to choose only one of them.